tests/monotouch-test/Network/NWProtocolIPOptionsTest.cs

```diff
@@ -45,7 +47,13 @@ public void Init ()
 			using (var endpoint = NWEndpoint.Create (host, "80")) {
 				connection = new NWConnection (endpoint, parameters);
 				connection.SetQueue (DispatchQueue.DefaultGlobalQueue); // important, else we will get blocked
-				connection.SetStateChangeHandler (ConnectionStateHandler);
+				connection.SetStateChangeHandler ((NWConnectionState state, NWError error) => {
+					try {
+						ConnectionStateHandler (state, error);
+					} catch (Exception e) {
+						ex = e;
+					}
+				});
 				connection.Start ();
```

Comment:
Will it be an error to  mess with the connection if the `ConnectionStateHandler` threw an exception?
if so, you could either put the follow three lines into `if (ex is null) {`.

Reply:
The state change handler will only be called *after* `connection.Start ()`, so there's no use in checking if ex is null before that.